untidy/contaminators.py

```diff
@@ -3,129 +3,197 @@
 import numpy as np
 import random
 
-""" Column-wise contaminators """
+""" Helpers """
 
-def get_data_dims(data):
+
+def get_random_indices(data, col_type="any", corruption_level=4):
     """
-    Check input data
+    Get random indeces to contaminate
 
-    Params
-    ------
-    data: pd.Series, pd.DataFrame
-        data to check
+    Parameters
+    ----------
+    data: pd.DataFrame
+        clean dataset
+    col_type: str, optional
+        'str', 'numeric' or 'any'. Type of columns to sample.
+    corruption_level int, optional
+        level of corruption, should be between 0 and 10, where 0 leaves the dataset as is, 10
+        is the highest level of contamination
 
     Returns
     -------
-    num_rows, num_cols: int(s)
-        integers specifying, respectively, the amount of rows and columns in data.
-        These values will be passed to *random_indeces* in order to generate the indeces to be contaminated
-
+    idxs: list
+        list of indeces to be contaminated
     """
-    if isinstance(data, pd.Series):
-        num_rows = data.shape[0]
-        num_cols = 1
 
-    elif isinstance(data, pd.DataFrame):
-        num_rows, num_cols = data.shape 
-        
+    # Define the number of datapoints to contaminate
+    num_obs = data.values.reshape(1, -1).shape[1]
+    prop_contaminated = np.linspace(0, 0.6, 11)[corruption_level]
+    num_contaminated = int(prop_contaminated * num_obs)
+
+    if isinstance(data, pd.DataFrame):
+        # Find columns to sample from
+        if col_type == "any":
+            cols_to_sample = range(len(data.columns))
+        elif col_type.startswith("str"):
+            cols_to_sample = [
+                list(data.columns).index(col)
+                for col in data.select_dtypes(include=["object", "category"]).columns
+            ]
+        elif col_type.startswith("num"):
+            cols_to_sample = [
+                list(data.columns).index(col)
+                for col in data.select_dtypes(include=["float64", "int64"]).columns
+            ]
```

Comment:
```suggestion
sampling_cols = lambda data, col_types: [list(data.columns).index(col) 
	for col in data.select_dtypes(include=col_types).columns]

type_dict = {'str':["object", "category"], 'num':["float64", "int64"]}

if col_type == 'all':
	cols_to_sample = range(len(data.columns))
else:
	cols_to_sample = sampling_cols(data, type_dict[col_type[0:3]]) 
```
This would make it a bit more concise and general, in case we need to work with additional types (e.g. dates) or combinations (only need 1 else instead of a new 'elif' every time

Reply:
I like this, applied, thanks!